packages/discovery-react-components/src/components/DocumentPreview/components/PdfViewer/PdfViewer.tsx

```diff
@@ -216,20 +238,28 @@ type CanvasInfo = {
   height: number;
   canvasWidth: number;
   canvasHeight: number;
+  proportion: number;
   viewport: PdfjsLib.PDFPageViewport;
 };
 
 function getCanvasInfo(
   loadedPage: PdfjsLib.PDFPageProxy | null | undefined,
-  scale: number
+  scale: number,
+  rootRef: RefObject<HTMLDivElement>
 ): CanvasInfo | null {
-  const canvasScale = window.devicePixelRatio ?? 1;
-  const viewport = loadedPage?.getViewport({ scale: scale * canvasScale });
-  if (viewport) {
+  const rootDimensions = rootRef?.current?.getBoundingClientRect();
+  if (loadedPage && rootDimensions) {
+    const canvasScale = window.devicePixelRatio ?? 1;
+    const width = rootDimensions.width * scale;
+    const pageWidth = loadedPage.view[2];
+    const proportion = rootDimensions.width / pageWidth;
+
+    const viewport = loadedPage.getViewport({
+      scale: (scale * canvasScale * width) / pageWidth
```

Comment:
This is weird. `width` is set to `rootDimensions.width * scale`, so then this becomes:

```js
scale * canvasScale * rootDimensions.width * scale / pageWidth
```

So multiplying by `scale` twice?

Given what was there in the previous code, it seems like what you want to do is:

```js
scale * canvasScale * proportion
```

Reply:
I'll have to dig a bit on this to make sure we're setting this properly to avoid any blurriness, but `scale * canvasScale * proportion` sounds right at the moment...